### ISSUE TITLE
Multi-GPU MoE (Ornith-1.5 / qwen35moe): plain decode + layer-split (PP) not runnable

### ISSUE THREAD

**Issue by DByte308** [Llaminar/llaminar#8]
# Multi-GPU MoE (Ornith-1.5 / qwen35moe): plain decode + layer-split (PP) not runnable

## Summary
On the current `develop` build of llaminar, a multi-GPU **MoE** model
(`qwen35moe`, e.g. Ornith-1.5-35B-A3B) loads and **tensor-parallelism (TP) works**,
but two requested paths are hard-blocked:

1. **Plain (non-MTP) ordinary generation is not installed for multi-GPU MoE.**
   Every non-MTP generation fails, and TP decode is additionally blocked by the
   column-parallel LM-head.
2. **Layer-split / pipeline-parallelism (PP) on a MoE model is not possible** in this
   build. Multi-device MoE orchestration refuses implicit cross-rank PP and routed-expert
   domains cannot be combined with `--pp-stage`.

This prevents the common two-way A/B (tensor split vs layer split) for MoE models that
works fine for dense models.

## Environment
- 4 x AMD MI50 (gfx906), 32 GiB VRAM each
- Host: Intel i7-6850K (AVX2 only) -> used `ghcr.io/llaminar/llaminar:develop-avx2`
- Model: `Ornith-1.5-35B-A3B-Q8_0.gguf`
  - GGUF arch `qwen35moe`: 40 main layers + 1 nextn/MTP block, 256 experts / 8 used,
    hybrid GDN + full-attention, 34.4 GB weights
- Backend: RCCL, `HSA_OVERRIDE_GFX_VERSION=9.0.6`

## 1) Plain generation not installed for multi-GPU MoE
Every non-MTP request (via HTTP `serve` or CLI `oneshot`/`benchmark`) fails decode:

```
Decode failed: Ordinary device generation: could not compose the complete captured
model and sampler; see the participant diagnostic
[DeviceGraphOrchestratorGenerator] Ordinary generation preparation failed:
Ordinary generation requires an admitted scalar sampler and complete local graph;
pipeline/overlay composition is not yet installed
```

`materializeOrdinaryDeviceGenerationLoopGraph()` rejects when either
`activeMainLogitsAreColumnParallel()` or `moe_overlay_epoch_execution_binding_` is set.
Consequences:
- With TP (apportioned experts) the overlay binding is active -> ordinary generation
  always refused. Only the **MTP** path (`--mtp --mtp-draft-tokens 1 --mtp-depth-policy fixed`)
  can generate.
- Even with MTP/plain, TP decode is gated on a local (non-column-parallel) LM head.
  `--mtp-terminal-head-policy mirrored-full-vocabulary` makes logits local, but the
  replicated ~248k-vocab head does not fit on 16 GB cards under TP (needs ~3.8 GB vs
  ~3.6 GB available at 16k context), so no working TP decode without some other layout.

Request: make ordinary generation work for multi-GPU MoE (or document that only
MTP-driven decode is supported), and relax the mirrored-head footprint for 16 GB cards.

## 2) Layer-split (PP) impossible on MoE
- `plan`/auto with `--only-strategies pp`:
  ```
  Implicit multi-device MoE authority normalization does not yet encode cross-rank
  TP or pipeline participants; provide one explicit ExpertOverlay domain plan ...
  ```
- Manual PP stages + explicit routed-expert domain:
  ```
  MoE routed-expert placement cannot be combined with --pp-stage in Phase 2:
  routed domains are same-layer expert roles, not PP layer ownership.
  ```

So a MoE model can only be spread via TP / ExpertOverlay (expert split), never layer
split. This matches the project's own benchmark harness
(`scripts/run_mtp_iteration_benchmark_matrix.sh`), which restricts `localtp_*`/`localpp_*`
to **dense** models and uses `routed_expert_tiered_*` (ExpertOverlay) for MoE only.

Request: decide whether layer-split PP is intended for MoE; if so, implement the
cross-rank MoE authority + PP composition, otherwise state it explicitly as unsupported
for MoE.

## Working reference (tensor-split + MTP)
```
llaminar2 benchmark -m Ornith-1.5-35B-Q8_0.gguf \
  --tp-devices rocm:0,rocm:1,rocm:2,rocm:3 --tensor-parallelism-degree 4 \
  --tp-scope rank_local --backend rccl \
  --mtp --mtp-draft-tokens 1 --mtp-depth-policy fixed \
  --activation-precision fp32 --kv-cache-precision fp16 ...
```
| context | prompt tok | prefill tok/s | decode tok/s (MTP) |
|--------:|-----------:|--------------:|-------------------:|
| short   | 227        | 611.4         | 83.2               |
| medium  | 908        | 634.2         | 69.7               |
| long    | 4086       | 724.4         | 74.8               |
